src/AspNetCoreModuleV2/AspNetCore/src/applicationinfo.cpp

```diff
@@ -221,8 +224,13 @@ APPLICATION_INFO::EnsureApplicationCreated()
                 hr = HRESULT_FROM_WIN32(ERROR_INVALID_FUNCTION);
                 goto Finished;
             }
+            hr = m_pfnAspNetCoreSetKeyValue(L"ExeLocation", exeLocation.c_str());
+            if (FAILED(hr))
+            {
+                goto Finished;
+            }
 
-            hr = m_pfnAspNetCoreCreateApplication(m_pServer, m_pConfiguration, &pApplication);
+            hr = m_pfnAspNetCoreCreateApplication(m_pServer, pHttpContext->GetApplication(), &pApplication);
```

Comment:
why not pass in the exe location string here so that you don't need to introduce another API?

Reply:
We don't want to have the executable path be a required parameter for creating the application. I'm adding a SetProperties function to the IApplication, which takes in arbitrary KVPs to be set on the application.